pytest.ini

```diff
@@ -21,6 +21,8 @@ filterwarnings =
     ignore:comparing unequal types not supported in 3\.x:DeprecationWarning
     ignore:the commands module has been removed in Python 3\.0:DeprecationWarning
     ignore:The 'new' module has been removed in Python 3\.0:DeprecationWarning
+# Do not mask this warning behind an error gh-15048
+    always:Creating an ndarray with automatic object dtype:DeprecationWarning
```

Comment:
Is this for debugging/flushing out, or to find mistakes? In release mode, we are not raising most warnings; or at least DeprecationWarnings, since in principle new ones could b e added by upstream.

Reply:
In what way was this warning masked behind an error before?

Reply:
On the dev branch, it would be raising as DeprecationWarning. In the poly code above, it would thus be caught by the `Exception` while in release, a warning would be shown.

Reply:
I'm not sure I understand. The `error` at the top of the filter starts off by turning all warnings into errors, but any that appear in the list are handled differently. I added a line to not raise the specific `DeprecationWarning` (as filtered by the message) as an error, but rather to always raise it as a warning. What is "release mode", this file only is active during testing.

Reply:
> it would thus be caught by the Exception while in release

It sounds like we should be searching for anything that possibly resembles `catch Exception`, and change it to:
```
catch Warning:
    raise
catch Exception:
    ...
```
I remember we had some C code change proposed to this recently